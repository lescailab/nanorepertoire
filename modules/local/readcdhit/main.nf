process READCDHIT {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/biopython:1.78':
        'ghcr.io/nibscbioinformatics/biopython:v1.78' }"

    input:
    tuple val(meta), path(clusters)


    output:
    tuple val(meta), path("*.summary"), emit: summary
    path '*.summary'                  , emit: summaryonly
    path "versions.yml"               , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    readcdout.py \
    ${clusters} \
    ${meta.id}_clusters.summary

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        biopython: \$(python -c "import Bio; print(Bio.__version__)")
    END_VERSIONS
    """
}
