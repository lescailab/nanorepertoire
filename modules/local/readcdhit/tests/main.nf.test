nextflow_process {

    name "Test Process READCDHIT"
    script "../main.nf"
    process "READCDHIT"
    config "./nextflow.config"
    tag "nanobodies"
    tag "modules"
    tag "modules_nfcore"
    tag "readcdhit"

    test("readcdhout") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test1', single_end:false ], // meta map
                    [file("https://github.com/lescailab/testdata/raw/master/nanobodies/test1.aa.clusters.clstr", checkIfExists: true)
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
