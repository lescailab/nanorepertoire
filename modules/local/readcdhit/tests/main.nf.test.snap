{
    "readcdhout": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test1",
                            "single_end": false
                        },
                        "test1_clusters.summary:md5,c65475d280e6c93dfad97004225a9c7d"
                    ]
                ],
                "1": [
                    "test1_clusters.summary:md5,c65475d280e6c93dfad97004225a9c7d"
                ],
                "2": [
                    "versions.yml:md5,dda93af047b2ef8b3e5dd648b6772255"
                ],
                "summary": [
                    [
                        {
                            "id": "test1",
                            "single_end": false
                        },
                        "test1_clusters.summary:md5,c65475d280e6c93dfad97004225a9c7d"
                    ]
                ],
                "summaryonly": [
                    "test1_clusters.summary:md5,c65475d280e6c93dfad97004225a9c7d"
                ],
                "versions": [
                    "versions.yml:md5,dda93af047b2ef8b3e5dd648b6772255"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-04T16:27:48.231144"
    }
}